File: VB/WebSite/App_Code/MyObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo



Public Class MyObject
	Inherits XPObject
	Public Sub New()
		MyBase.New()
	End Sub

	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	Public Overrides Sub AfterConstruction()
		MyBase.AfterConstruction()
	End Sub

	Private _Parent As MyObject
	<Association("Parent-Children")> _
	Public Property Parent() As MyObject
		Get
			Return _Parent
		End Get
		Set(ByVal value As MyObject)
			SetPropertyValue("Parent", _Parent, value)
		End Set
	End Property

	Protected text_Renamed As String
	Public Property Text() As String
		Get
			Return text_Renamed
		End Get
		Set(ByVal value As String)
			SetPropertyValue(Of String)("Title", text_Renamed, value)
		End Set
	End Property

	<Association("Parent-Children")> _
	Public ReadOnly Property Children() As XPCollection(Of MyObject)
		Get
			Return GetCollection(Of MyObject)("Children")
		End Get
	End Property

	'public override string ToString() {
	'    return Title;
	'}
End Class

File: CS/WebSite/App_Code/MyObject.cs
using System;
using DevExpress.Xpo;



public class MyObject : XPObject {
    public MyObject()
        : base() { }

    public MyObject(Session session)
        : base(session) { }

    public override void AfterConstruction() {
        base.AfterConstruction();
    }

    MyObject _Parent;
    [Association("Parent-Children")]
    public MyObject Parent {
        get { return _Parent; }
        set { SetPropertyValue("Parent", ref _Parent, value); }
    }

    protected String text;
    public String Text {
        get { return text; }
        set { SetPropertyValue<String>("Title", ref text, value); }
    }

    [Association("Parent-Children")]
    public XPCollection<MyObject> Children {
        get { return GetCollection<MyObject>("Children"); }
    }

    //public override string ToString() {
    //    return Title;
    //}
}

